system/src/system_cloud_internal.cpp

```diff
@@ -627,15 +628,24 @@ void Spark_Signal(bool on, unsigned, void*)
 
 size_t system_interpolate_cloud_server_hostname(const char* var, size_t var_len, char* buf, size_t buf_len)
 {
-	if (var_len==2 && memcmp("id", var, 2)==0)
-	{
-		String deviceID = spark_deviceID();
-		if (buf_len>deviceID.length()) {
-			memcpy(buf, deviceID.c_str(), deviceID.length());
-			return deviceID.length();
-		}
-	}
-	return 0;
+    if (var_len==2 && memcmp("id", var, 2)==0)
+    {
+        String deviceID = spark_deviceID();
+        size_t id_len = deviceID.length();
+
+        SystemVersionInfo sys_ver;
+        system_version_info(&sys_ver, nullptr);
+        uint8_t mv = BYTE_N(sys_ver.versionNumber, 3);
+        String majorVer = String::format(".v%d", mv);
```

Comment:
Nit: ideally we shouldn't use `wiring` within DeviceOS whenever possible.

Reply:
I considered replacing what was there already, but then thought... hey this makes sure we don't break String :D and probably 5 less complaints from Codacy about strlen() and strncpy().

Reply:
> hey this makes sure we don't break String

We have tests for that :smirk: 

>  5 less complaints from Codacy about strlen() and strncpy().

Just ignore them, those specifically are annoying and don't bring any value.